src/secret.rs

```diff
@@ -152,6 +155,28 @@ mod use_std {
             Self(secret)
         }
 
+        /// Generate a new secret key from a mnemonic phrase and its derivation path.
+        /// Both are passed as `&str`. If you want to manually create a `DerivationPath`
+        /// and `Mnemonic`, use [`SecretKey::new_from_mnemonic`].
+        /// The derivation path is a list of integers, each representing a child index.
+        pub fn new_from_mnemonic_phrase_with_path(phrase: &str, path: &str) -> Result<Self, Error> {
+            let mnemonic = Mnemonic::<W>::new_from_phrase(phrase)?;
+
+            let path = DerivationPath::from_str(path)?;
+
+            let derived_priv_key = mnemonic.derive_key(path, None)?;
+            let key: &coins_bip32::prelude::SigningKey = derived_priv_key.as_ref();
+            Ok(unsafe { SecretKey::from_slice_unchecked(key.to_bytes().as_ref()) })
```

Comment:
Forgot to mention.

Please, add a `Safety` comment explaining why this usage is safe

For reference, the safety constraints of this functions are documented in:
https://docs.rs/fuel-crypto/latest/fuel_crypto/struct.SecretKey.html#safety-2

You need to put a comment on why this slice will always be of the expected len